File: april_week_2_spaced_repetition/reverseLinkedList.js
function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
}

var reverseList = function(head) {
    let [prev, curr] = [null, head];
    
    while (curr) {
        let temp = curr;
        curr = curr.next;
        temp.next = prev;
        prev = temp;
    }
    return prev;
};

console.log(reverseList(1))

File: swift/reverseLinkedList.swift

public class ListNode {
    public var val: Int
    public var next: ListNode?
    public init() { self.val = 0; self.next = nil; }
    public init(_ val: Int) { self.val = val; self.next = nil; }
    public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
}
 
class Solution {
    func reverseList(_ head: ListNode?) -> ListNode? {
        var prev: ListNode?
        var curr = head

        while let node = curr {
            var next = node.next
            node.next = prev
            prev = node
            curr = next
        }
        return prev
    }
}